### PR TITLE
Remove assert(false) and make code better

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [x] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by Dronplane — APPROVED** [review]
LGTM

File: arangod/IResearch/ViewSnapshot.cpp [hunk 1]
```diff
@@ -87,7 +87,8 @@ bool ViewSnapshotCookie::compute(bool sync, std::string_view name) {
 
 void ViewSnapshotCookie::add(DataSourceId cid,
                              irs::directory_reader&& reader) noexcept {
-  _readers.reserve(_readers.size() + reader.size());
+  _readers.reserve(_readers.size() +
+                   std::max(reader.size(), _readers.size() / 2));
```
[the remark this answers is not in the document]
Comment (MBkkt) [reply arangod/IResearch/ViewSnapshot.cpp:91]: Just make it better.
In general we want to achieve two goals, for _readers here:
1) no more than single allocation
2) this function called in cycle, so we don't want make more than log(cycle size) allocation

I think we have alternative approach:
Save all reader
Compute size
Reserve buffer
Compute other stuff

But its a little more complicated?

[the remark this answers is not in the document]
Comment (Dronplane) [reply arangod/IResearch/ViewSnapshot.cpp:91]: Ah no, growing by capacity is wrong of course

[the remark this answers is not in the document]
Comment (MBkkt) [reply arangod/IResearch/ViewSnapshot.cpp]: I think fixed?

[the remark this answers is not in the document]
Comment (Dronplane) [reply arangod/IResearch/ViewSnapshot.cpp]: Makes sense 👍

**Pull request merged**